sqlglot/tokens.py

```diff
@@ -820,17 +832,29 @@ def _scan_comment(self, comment_start):
         if comment_start not in self._COMMENTS:
             return False
 
+        comment_start_line = self._line
+        comment_start_size = len(comment_start)
         comment_end = self._COMMENTS[comment_start]
 
         if comment_end:
             comment_end_size = len(comment_end)
 
             while not self._end and self._chars(comment_end_size) != comment_end:
                 self._advance()
+
+            self._comment = self._text[comment_start_size : -comment_end_size + 1]
             self._advance(comment_end_size - 1)
         else:
             while not self._end and self.WHITE_SPACE.get(self._peek) != TokenType.BREAK:
                 self._advance()
+            self._comment = self._text[comment_start_size:]
+
+        # Leading comment is attached to the succeeding token, whilst trailing comment to the preceding. If both
+        # types of comment can be attached to a token, the trailing one is discarded in favour of the leading one.
+        if comment_start_line == self._prev_token_line and self._prev_token_comment is None:
+            self.tokens[-1].comment = self._comment
```

Comment:
don't you always attach to the preceding? how do you do the inverse case?

Reply:
I cache the comment internally and add it to the next token inside add.